src/LAMMPS.jl

```diff
@@ -177,240 +245,362 @@ function get_natoms(lmp::LMP)
     Int64(API.lammps_get_natoms(lmp))
 end
 
-function dtype2type(dtype::API._LMP_DATATYPE_CONST)
-    if dtype == API.LAMMPS_INT
-        type = Ptr{Int32}
-    elseif dtype == API.LAMMPS_INT_2D
-        type = Ptr{Ptr{Int32}}
-    elseif dtype == API.LAMMPS_INT64
-        type = Ptr{Int64}
-    elseif dtype == API.LAMMPS_INT64_2D
-        type = Ptr{Ptr{Int64}}
-    elseif dtype == API.LAMMPS_DOUBLE
-        type = Ptr{Float64}
-    elseif dtype == API.LAMMPS_DOUBLE_2D
-        type = Ptr{Ptr{Float64}}
-    elseif dtype == API.LAMMPS_STRING
-        type = Ptr{Cchar}
-    else
-        @assert false "Unknown dtype: $dtype"
-    end
-    return type
+function _lammps_string(ptr::Ptr)
+    ptr == C_NULL && error("Wrapping NULL-pointer!")
+    return Base.unsafe_string(ptr)
 end
 
-"""
-    extract_global(lmp, name, dtype=nothing)
-"""
-function extract_global(lmp::LMP, name, dtype=nothing)
-    if dtype === nothing
-        dtype = API.lammps_extract_global_datatype(lmp, name)
-    end
-    dtype = API._LMP_DATATYPE_CONST(dtype)
-    type = dtype2type(dtype)
+function _lammps_wrap(ptr::Ptr{<:Real}, shape::Integer; own=false)
+    ptr == C_NULL && error("Wrapping NULL-pointer!")
+    return Base.unsafe_wrap(Array, ptr, shape, own=own)
+end
 
-    ptr = API.lammps_extract_global(lmp, name)
-    ptr = reinterpret(type, ptr)
+function _lammps_wrap(ptr::Ptr{<:Ptr{T}}, shape::NTuple{2}; own=false) where T
```

Comment:
I don't particularly like the naming< we are in the LAMMPS module, so adding LAMMPS feels superfluous.

`unsafe_wrap` or just wrap

Reply:
I think It's best to just roll these two functions into one again but call it `_extract` instead of `wrap`.

we basically allways do
```julia
copy ? _lammps_copy(ptr, length) : _lammps_wrap(ptr, length)
```
afterall.

Reply:
I don't think we need `_copy`

Is there a material benefit over. `copy(unsafe_wrap())`

I do expect advanced users being exposed to wrap, since they might use the LAMMPS API directly